### PR TITLE
Implemented the SELFDESTRUCT Opcode

### PR DESCRIPTION
**Pull request by voith**
### What was wrong?

depends on #280 and #316

### How was it fixed?
Implemented the `SELFDESTRUCT` Opcode. 

### Note
Although the frontier version had named this opcode as `SUICIDE`, a decision was made in [discord](https://discord.com/channels/595666850260713488/856943705209569291/869592788096917504) that as  per EIP-6 we will rename it to `SELFDESTRUCT`.

#### Cute Animal Picture

Screenshot 2021-08-26 at 7 17 06 PM

Pic credit: [africanoverlandtours.com](https://www.africanoverlandtours.com/overland-africa-blog/)

### CODE REVIEW

**Comment by voith** [comment]
@SamWilsn This is ready for review.

File: src/ethereum/frontier/vm/instructions/system.py [hunk 1]
```diff
@@ -260,3 +265,38 @@ def callcode(evm: Evm) -> None:
...
+    # Next, Zero the balance of the address being deleted (must come after
+    # sending to beneficiary in case the contract named itself as the
+    # beneficiary).
+    set_account_balance(evm.env.state, originator, U256(0))
+
+    # gas refund
+    evm.refund_counter += REFUND_SELF_DESTRUCT
+
+    # register account for deletion
+    evm.accounts_to_delete.add(beneficiary)
```
Comment (SamWilsn) [inline src/ethereum/frontier/vm/instructions/system.py:299]: What happens when you `call` into a contract, which self-destructs, twice in the same transaction?

Comment (voith) [reply src/ethereum/frontier/vm/instructions/system.py:299]: resolved on discord.

Comment (voith) [inline src/ethereum/frontier/vm/instructions/system.py:299]: This is incorrect. I should register the `originator` for delete and not the beneficiary. 
Sadly no test caught this. I'm looking for tests at the moment that will catch this bug.

**Pull request merged**